### PR TITLE
[DOC] update doc for 2408 release [skip CI]

### PR DESCRIPTION
**Pull request by nvliyuan**
This PR updates download docs for 24.08.0 release.

### CODE REVIEW

File: docs/download.md [hunk 1]
```diff
@@ -43,15 +43,17 @@ The plugin is tested on the following architectures:
 		Apache Spark 3.3.0, 3.3.1, 3.3.2, 3.3.3, 3.3.4
 		Apache Spark 3.4.0, 3.4.1, 3.4.2, 3.4.3
 		Apache Spark 3.5.0, 3.5.1
+		Apache Spark 4.0.0
```
Comment (NvTimLiu) [inline docs/download.md:46]: v24.08.0 release will not contain the 4.0.0 shim, and Apache spark-4.0.0 is not released

Comment (nvliyuan) [reply docs/download.md]: updated, thx

File: docs/download.md [hunk 2]
```diff
@@ -43,15 +43,17 @@ The plugin is tested on the following architectures:
...
 	
 	Supported Databricks runtime versions for Azure and AWS:
 		Databricks 11.3 ML LTS (GPU, Scala 2.12, Spark 3.3.0)
 		Databricks 12.2 ML LTS (GPU, Scala 2.12, Spark 3.3.2)
 		Databricks 13.3 ML LTS (GPU, Scala 2.12, Spark 3.4.1)
+		Databricks 14.3 ML LTS (GPU, Scala 2.12, Spark 3.5.0)
```
[the remark this answers is not in the document]
Comment (nvliyuan) [reply docs/download.md]: updated, thx

**Review by NvTimLiu — DISMISSED** [review]
LGTM, +1

**Comment by NvTimLiu** [comment]
build

**Comment by nvliyuan** [comment]
@jlowe @sameerz can you please help review?

File: docs/download.md [hunk 3]
```diff
@@ -84,19 +84,20 @@ CUDA 11.8 through CUDA 12.0.
...
 ### Release Notes
-* Improve support for Unity Catalog on Databricks
-* Added support for parse_url PATH
-* Added support for array_filter
-* Added support for Spark 3.4.3
+* Support timezones with daylight savings shifts
+* Improve metrics in Spark UI
+* Refact parquet decode microkernels and support load balancing RLE runs
+* Support dynamic scan filtering
+* Improve UCX shuffle
```
Comment (jlowe) [inline docs/download.md:100]: May want to add:
- Improve get_json performance

Comment (nvliyuan) [reply docs/download.md:101]: added

**Comment by nvliyuan** [comment]
build

**Review by jlowe — APPROVED** [review]

**Pull request merged**